src/protocols/secure_channel/tests/TestPASESession.cpp

```diff
@@ -138,25 +172,47 @@ void SecurePairingHandshakeTestCommon(nlTestSuite * inSuite, void * inContext, P
 
     gLoopback.mSentMessageCount = 0;
 
-    NL_TEST_ASSERT(inSuite, pairingCommissioner.MessageDispatch().Init(&gTransportMgr) == CHIP_NO_ERROR);
-    NL_TEST_ASSERT(inSuite, pairingAccessory.MessageDispatch().Init(&gTransportMgr) == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite,
+                   pairingCommissioner.MessageDispatch().Init(ctx.GetExchangeManager().GetReliableMessageMgr(), &gTransportMgr) ==
+                       CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite,
+                   pairingAccessory.MessageDispatch().Init(ctx.GetExchangeManager().GetReliableMessageMgr(), &gTransportMgr) ==
+                       CHIP_NO_ERROR);
+
+    ExchangeContext * contextCommissioner = ctx.NewExchangeToLocal(&pairingCommissioner);
+
+    if (gLoopback.mNumMessagesToDrop != 0)
+    {
+        pairingCommissioner.MessageDispatch().SetPeerAddress(PeerAddress(Type::kUdp));
+        pairingAccessory.MessageDispatch().SetPeerAddress(PeerAddress(Type::kUdp));
+
+        ReliableMessageMgr * rm     = ctx.GetExchangeManager().GetReliableMessageMgr();
+        ReliableMessageContext * rc = contextCommissioner->GetReliableMessageContext();
+        NL_TEST_ASSERT(inSuite, rm != nullptr);
+        NL_TEST_ASSERT(inSuite, rc != nullptr);
+
+        rc->SetConfig({
+            1, // CHIP_CONFIG_RMP_DEFAULT_INITIAL_RETRY_INTERVAL
+            1, // CHIP_CONFIG_RMP_DEFAULT_ACTIVE_RETRY_INTERVAL
+        });
+        gLoopback.mContext = &ctx;
+    }
 
     NL_TEST_ASSERT(inSuite,
                    ctx.GetExchangeManager().RegisterUnsolicitedMessageHandlerForType(
                        Protocols::SecureChannel::MsgType::PBKDFParamRequest, &pairingAccessory) == CHIP_NO_ERROR);
 
-    ExchangeContext * contextCommissioner = ctx.NewExchangeToLocal(&pairingCommissioner);
-
     NL_TEST_ASSERT(inSuite,
                    pairingAccessory.WaitForPairing(1234, 500, (const uint8_t *) "saltSALT", 8, 0, &delegateAccessory) ==
                        CHIP_NO_ERROR);
     NL_TEST_ASSERT(inSuite,
                    pairingCommissioner.Pair(Transport::PeerAddress(Transport::Type::kBle), 1234, 0, contextCommissioner,
                                             &delegateCommissioner) == CHIP_NO_ERROR);
 
-    NL_TEST_ASSERT(inSuite, gLoopback.mSentMessageCount == 5);
+    NL_TEST_ASSERT(inSuite, gLoopback.mSentMessageCount >= 5);
```

Comment:
Can't we test `gLoopback.mSentMessageCount - gLoopback.mDroppedMessageCount == 5`?

Reply:
Actually, it counts the standalone acks as well. And, tough to say if acks will come as standalone or be part of a reply...

Reply:
Ah, ok.  Might be worth a comment.